File: content/api-documentation/how-to/orders/examples/streaming.js
const financefeast = require('@financefeasthq/financefeast-trade-api')
const financefeast = new financefeast()

// Prepare the websocket connection and subscribe to trade_updates.
financefeast.websocket.onConnect(function () {
    console.log("Connected")
    client.subscribe(['trade_updates'])
    setTimeout(() => {
        client.disconnect()
    }, 30 * 1000)
})
financefeast.websocket.onDisconnect(() => {
    console.log("Disconnected")
})
financefeast.websocket.onStateChange(newState => {
    console.log(`State changed to ${newState}`)
})

// Handle updates on an order you've given a Client Order ID.
const client_order_id = 'my_client_order_id'
financefeast.websocket.onOrderUpdate(orderData => {
    const event = orderData['event']
    if (orderData['order']['client_order_id'] == client_order_id) {
        console.log(`Update for ${client_order_id}. Event: ${event}.`)
    }
})

// Start listening for updates.
financefeast.websocket.connect()

File: content/api-documentation/how-to/orders/examples/streaming.go
package main

import (
	"fmt"

	"github.com/financefeasthq/financefeast-trade-api-go/financefeast"
    "github.com/financefeasthq/financefeast-trade-api-go/stream"
)

func init() {
	financefeast.SetBaseUrl("https://paper-api.financefeast.markets")
}

func main() {
	if err := stream.Register(financefeast.TradeUpdates, tradeHandler); err != nil {
        panic(err)
    }
}

func tradeHandler(msg interface{}) {
	tradeupdate := msg.(financefeast.TradeUpdate)
	if tradeupdate.Order.ClientOrderID == "my_client_order_id" {
		fmt.Printf("Update for {}. Event: {}.\n", tradeupdate.Order.ClientOrderID, tradeupdate.Event)
	}
}